heritability/rhemc_pipeline.py

```diff
@@ -0,0 +1,1334 @@
+            saige_file = convert_pheno_id_to_potential_saige(pheno_id)
+            dct_out.update({pheno_id: (pheno_file, saige_file, ancestries)})
+        return dct_out
+    else:
+        phenotype_id = manifest.phenotype_id.collect()
+        pheno_file = manifest.pheno_file.collect()
+        saige_file = manifest.saige_file.collect()
+        anc_vec = [[anc for anc in anclist if anc in ancestries] 
+                for anclist in manifest.pop_split.collect()]
+
+        pheno_to_len = {phen: len(anclist) for phen, _, _, anclist in zip(phenotype_id, pheno_file, saige_file, anc_vec)}
+        zipped_vals = zip(phenotype_id, pheno_file, saige_file, anc_vec)
+        dct_out = {id: (file, saigefile, anclist) for id, file, saigefile, anclist in zipped_vals if len(anclist) > 0}
+
+        if n_include is not None:
+            trait_types = list(manifest.aggregate(hl.agg.collect_as_set(manifest.trait_type)))
+            list_of_pheno_lists = []
+            for trait_type in trait_types:
+                corresponding_phenos = manifest.filter(manifest.trait_type == trait_type).phenotype_id.collect()
+                corr_phenos_slim = [pheno for pheno in corresponding_phenos if pheno_to_len[pheno] > 0]
+                list_of_pheno_lists.append(corr_phenos_slim[0:min(n_include, len(corr_phenos_slim))])
+            traits_to_keep = [item for pheno_list in list_of_pheno_lists for item in pheno_list]
+            return {key: dct_out[key] for key in traits_to_keep}
+        elif specific_pheno is not None:
+            if not all([x in dct_out.keys() for x in specific_pheno]):
+                missing_pheno = ', '.join([x for x in specific_pheno if x not in dct_out.keys()])
+                raise ValueError('ERROR: the following phenotypes were not found: ' + missing_pheno)
+            return {key: dct_out[key] for key in specific_pheno}
+        else:
+            return dct_out
+
+
+def _read_pheno_rowtable():
+    """ Reads the row Hail Table from the FULL phenotype MatrixTable.
+    """
+    return _read_pheno_data(False,load_full=True).rows()
+
+
+def _read_pheno_data(checkpoint, load_full=False):
+    """ Reads phenotype data, extending the stock phenotype matrix table
+    by using a single phenotype ID and keeping only a single row, entry, and column field
+    unless load_full is set to True.
+
+    Returns
+    -------
+    `MatrixTable`
+    """
+    checkpoint_phenos = mtcheckpoint + 'filtered_phenotype_data_temporary.mt'
+    if load_full:
+        return hl.read_matrix_table(get_ukb_pheno_mt_path())
+    else:
+        if hl.hadoop_exists(checkpoint_phenos):
+            pheno_mt_string = hl.read_matrix_table(checkpoint_phenos)
+        else:
+            pheno_mt = hl.read_matrix_table(get_ukb_pheno_mt_path())
+            pheno_mt_string = pheno_mt.annotate_cols(phenotype_id = construct_phenotype_id(pheno_mt))
+            pheno_mt_string = pheno_mt_string.key_cols_by('phenotype_id')
```

Comment:
`key_cols_by` can also take a keyword argument so you can combine these as well

Reply:
Done